tests/push/test_presentable_names.py

```diff
@@ -68,27 +70,27 @@ class PresentableNamesTestCase(unittest.HomeserverTestCase):
 
     def _calculate_room_name(
         self,
-        events: StateMap[dict],
+        events: Iterable[Tuple[Tuple[str, str], dict]],
         user_id: str = "",
         fallback_to_members: bool = True,
         fallback_to_single_member: bool = True,
-    ):
-        # This isn't 100% accurate, but works with MockDataStore.
-        room_state_ids = {k[0]: k[0] for k in events}
+    ) -> Optional[str]:
+        # Encode the state key into the event ID.
+        room_state_ids = {k[0]: "|".join(k[0]) for k in events}
```

Comment:
To check: this is now a map from `(type, state_key)` to `f"{type}|{state_key}"`? E.g. `("m.room.member", "a") -> "m.room.member|a"`.

Reply:
Yeah. We could do more tracking and generate integer event IDs are something but that seemed like a faff.